engines/myst3/myst3.cpp

```diff
@@ -1487,6 +1491,28 @@ bool Myst3Engine::canLoadGameStateCurrently() {
 	return _interactive;
 }
 
+void Myst3Engine::tryAutoSaving() {
+	if (!canLoadGameStateCurrently()) {
+		return; // Can't save right now, try again on the next frame
+	}
+
+	_lastSaveTime = _system->getMillis();
+
+	if (!_state->isAutoSaveAllowed()) {
+		return; // Can't autosave ever, try again after the next autosave delay
+	}
+
+	Common::String saveName = "Autosave";
+	Common::String fileName = saveName + ".M3S";
+
+	// Save the state and the thumbnail
+	Common::OutSaveFile *save = getSaveFileManager()->openForSaving(fileName);
+	_state->setSaveDescription(saveName);
```

Comment:
I don't think this is correct.
1. The user saves his game with a custom name
2. Autosave happens
3. Suddenly when the user tries to save, his game is named "autosave"